Assets/Scripts/Attacks/PlayerBasicAttack.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Assets.Scripts.Attacks
+{
+    public class PlayerBasicAttack : MonoBehaviour
+    {
+        private GameObject arrowPrefab;
+        private GameObject arrowSpawningPosition;
+        private bool isShootingRight = true;
+        private float cooldown = 0.0f;
+
+        private void Awake()
+        {
+            arrowPrefab = (GameObject)Resources.Load("Prefabs/Arrow", typeof(GameObject));
+            arrowSpawningPosition = GameObject.FindGameObjectWithTag("ArrowSpawningPosition");
+        }
+
+        private void Update()
+        {
+            cooldown -= Time.deltaTime;
+            if (Input.GetKeyDown(KeyCode.Space) && cooldown < 0)
+            {
+                isShootingRight = GameObject.FindGameObjectWithTag("Player").GetComponent<PlayerMovement>().isFacingRight;
```

Comment:
GameObject.FindGameObjectWithTag("Player").GetComponent<PlayerMovement>() is a heavy opertaion, that gets called every frame in the Update function. Store the PlayerMovement as a property of this script, so this will be called only one time in the Awake function.

Reply:
done